tests/block1tests.c

```diff
@@ -21,62 +21,119 @@
 #include "internals.h"
 #include "liblwm2m.h"
 
+static const char *URI = "/1/2/3";
+static const uint16_t BLOCK_SIZE = 5;
 
-static void handle_12345(lwm2m_block_data_t ** blk1,
-                                  const char * uri) {
+static uint8_t handle_12345(lwm2m_block_data_t **blk1, uint8_t *resultBuffer, size_t *resultLen) {
     uint8_t *buffer = (uint8_t *)"12345";
-    size_t bsize;
-    uint8_t *resultBuffer = NULL;
+    size_t bufferLength = strlen(buffer);
+    const int BLOCK_NUM = 0;
+    const bool BLOCK_MORE = true;
 
-    uint8_t st = coap_block1_handler(blk1, uri, buffer, 5, 5, 0, true, &resultBuffer, &bsize);
-    CU_ASSERT_EQUAL(st, COAP_231_CONTINUE)
-    CU_ASSERT_PTR_NULL(resultBuffer)
+    return coap_block1_handler(blk1, URI, buffer, bufferLength, BLOCK_SIZE, BLOCK_NUM, BLOCK_MORE, resultBuffer,
+                               resultLen);
 }
 
-static void handle_67(lwm2m_block_data_t ** blk1,
-                                  const char * uri) {
+static uint8_t handle_67(lwm2m_block_data_t **blk1, uint8_t *resultBuffer, size_t *resultLen) {
     uint8_t *buffer = (uint8_t *)"67";
-    size_t bsize;
-    uint8_t *resultBuffer = NULL;
+    size_t bufferLength = strlen(buffer);
 
-    uint8_t st = coap_block1_handler(blk1, uri, buffer, 2, 5, 1, false, &resultBuffer, &bsize);
-    CU_ASSERT_EQUAL(st, NO_ERROR)
-    CU_ASSERT_PTR_NOT_NULL(resultBuffer)
-    CU_ASSERT_EQUAL(bsize, 7)
-    CU_ASSERT_NSTRING_EQUAL(resultBuffer, "1234567", 7)
+    const int BLOCK_NUM = 1;
+    bool BLOCK_MORE = false;
+    return coap_block1_handler(blk1, URI, buffer, bufferLength, BLOCK_SIZE, BLOCK_NUM, BLOCK_MORE, resultBuffer,
+                               resultLen);
 }
 
-
 static void test_block1_nominal(void)
 {
     lwm2m_block_data_t * blk1 = NULL;
 
-    handle_12345(&blk1, "/1/2/3");
-    handle_67(&blk1, "/1/2/3");
+    uint8_t *resultBuffer = NULL;
+    size_t resultLen;
+
+    uint8_t status = handle_12345(&blk1, resultBuffer, &resultLen);
+    CU_ASSERT_EQUAL(status, COAP_231_CONTINUE)
+    CU_ASSERT_PTR_NULL(resultBuffer)
+
+    status = handle_67(&blk1, &resultBuffer, &resultLen);
```

Comment:
```suggestion
    status = handle_67(&blk1, resultBuffer, &resultLen);
```

To avoid errors go unnoticed due to results carrying over from old invocations - how about assigning resultBuffer/resultLen/etc. non-test-passing values before invoking handle_67()?

```suggestion
    resultBuffer = NULL;
    resultLen = 0;
    status = handle_67(&blk1, resultBuffer, &resultLen);
```

(same goes for some other places)